sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/src/AzureKeyVaultConfigurationProvider.cs

```diff
@@ -125,68 +129,22 @@ await foreach (var secret in secretPages.ConfigureAwait(false))
             }
         }
 
-        private void SetData(Dictionary<string, LoadedSecret> loadedSecrets, bool fireToken)
-        {
-            var data = new Dictionary<string, LoadedSecret>(loadedSecrets.Count, StringComparer.OrdinalIgnoreCase);
-
-            // The loadesSecrets dictionary contains LoadedSecrets objects that has
-            // the configuration value and the configuration key (created by the
-            // KeyVaultSecretManager object). It is possible that multiple
-            // LoadedSecrets objects uses the same configuration key. This loop
-            // takes the latest updated value for each key.
-            foreach (var secretItem in loadedSecrets)
-            {
-                string key = secretItem.Value.Key;
-
-                if (data.TryGetValue(key, out LoadedSecret currentSecret))
-                {
-                    if (secretItem.Value.Updated > currentSecret.Updated)
-                    {
-                        data[key] = secretItem.Value;
-                    }
-                }
-                else
-                {
-                    data.Add(key, secretItem.Value);
-                }
-            }
-
-            Data = data.ToDictionary(d => d.Key, v => v.Value.Value);
-            if (fireToken)
-            {
-                OnReload();
-            }
-        }
-
         /// <inheritdoc/>
-        public void Dispose()
+        void IDisposable.Dispose()
```

Comment:
Are people meant to be able to derive this class now? If so, this should probably use the protected `Disposing(bool)` pattern.